sound/soc/sof/intel/hda.h

```diff
@@ -386,6 +387,11 @@ static inline struct hda_bus *sof_to_hbus(struct snd_sof_dev *s)
 	return &hda->hbus;
 }
 
+struct sof_intel_hda_stream {
+	struct hdac_ext_stream hda_stream;
+	struct sof_intel_stream stream;
+};
```

Comment:
it's getting complicated, we have sof_intel_hda_stream, hdac_ext_stream and sof_intel_stream.
How many stream definitions do we need? You really lost me here.

Reply:
I didn't want to modify struct sof_intel_hda_stream or struct hdac_ext_stream because they're also used by non-SOF HDA drivers. That's why I had to add an SOF-specific stream struct. But if we want to modify one of those structs instead, we can do that too.

Reply:
maybe call the member sof_stream? 'stream' makes it hard to search for members and it's used too often for local variables.